docs/developers/instantiate-testnet.md

```diff
@@ -0,0 +1,244 @@
+downloading files, compiling and monitoring the node:
+
+```sh
+sudo apt install curl tar wget clang pkg-config libssl-dev jq \
+build-essential git make ncdu -y
+```
+
+### Install Golang
+
+Golang will be installed on this machine in order for us to be able to build
+the necessary binaries for running the bridge node. For Golang specifically,
+it is needed to be able to compile Celestia Application.
+
+```sh
+ver="1.17.2"
+cd $HOME
+wget "https://golang.org/dl/go$ver.linux-amd64.tar.gz"
+sudo rm -rf /usr/local/go
+sudo tar -C /usr/local -xzf "go$ver.linux-amd64.tar.gz"
+rm "go$ver.linux-amd64.tar.gz"
+```
+
+Now we need to add the `/usr/local/go/bin` directory to `$PATH`:
+
+```sh
+echo "export PATH=$PATH:/usr/local/go/bin:$HOME/go/bin" >> $HOME/.bash_profile
+source $HOME/.bash_profile
+```
+
+To check if Go was installed correctly run:
+
+```sh
+go version
+```
+
+Output should be the version installed:
+
+```console
+go version go1.17.2 linux/amd64
+```
+
+## Installation
+
+Run the following:
+
+```sh
+cd $HOME
+git clone https://github.com/celestiaorg/celestia-app.git
+cd celestia-app/
+git checkout tags/v0.4.0 -b v0.4.0
+make install
+```
+
+To check if the binary was successfully compiled you can run the binary
+using the `--help` flag:
+
+```sh
+celestia-appd --help
+```
+
+You should see a similar output:
+
+```console
+Stargate CosmosHub App
+
+Usage:
+  celestia-appd [command]
+
+Available Commands:
+*
+*
+*
+Use "celestia-appd [command] --help" for more information about a command.
+```
+
+## Spin Up A Celestia Testnet
+
+If you want to spin up a quick testnet with your friends, you can follow these steps.
+Unless otherwise noted, every step must be done by everyone who wants to
+participate in this testnet.
+
+### Optional: Reset Working Directory
+
+If you have already initialized a working directory for `celestia-appd` in the past,
+you must clean up before reinitializing a new directory. You can do so by running
+the following command:
+
+```sh
+celestia-appd unsafe-reset-all
+```
+
+### Initialize A Working Directory
+
+Run the following command:
+
+```sh
+VALIDATOR_NAME=validator1
+CHAIN_ID=testnet
+celestia-appd init $VALIDATOR_NAME --chain-id $CHAIN_ID
+```
+
+* The value we will use for `$VALIDATOR_NAME` is `validator1` but you should choose
+  your own node name.
+* The value we will use for `$CHAIN_ID` is `testnet`. The `$CHAIN_ID` must
+  remain the same for everyone participating in this network.
+
+### Create A New Key
+
+Next, run the following command:
+
+```sh
+KEY_NAME=validator
+celestia-appd keys add $KEY_NAME 
+```
+
+This will create a new key, with a name of your choosing.
+Save the output of this command somewhere; you'll need
+the address generated here later. Here, we set the value of our
+key `$KEY_NAME` to `validator` for demonstration.
+
+### Add Genesis Account KeyName
+
+Run the following command:
+
+```sh
+celestia-appd add-genesis-account $KEY_NAME [amount]
```

Comment:
Get the following error. cc @evan-forbes 
```console
$ celestia-appd add-genesis-account $KEY_NAME 10000000000000000000000000uceles
Error: failed to get address from Keybase: validator.info: key not found [celestiaorg/cosmos-sdk@v1.0.1-sdk-v0.44.2/crypto/keyring/keyring.go:465]
```

Reply:
We're not though. See the above instructions. It's using the default keyring.

Reply:
I missed the step to adding `--keyring-backend test` but can confirm it worked during the dry run with @evan-forbes last Friday. Can add it to the instructions.

Reply:
Updated